### ISSUE TITLE
--output for specific files at the end of layers [Feature Request]

### ISSUE THREAD

**Issue by rufreakde** [GoogleContainerTools/kaniko#1832]
# Description
When using kaniko inside of pipelines it would be very benificial to have some way to export files from a folder similar as to how you can export an image with `--tarPath` but without any requironments for a `--destination`. 

# Benefits
- Simpler pipeline configuration with workspaces and file sharing between stages/steps
- Simpler local test runs where you can directly check contents

# Difference to docker
Docker build has a rudamental --output command which can only export the --target layer but I think it would be simple to just build the complete image and add as many --output commands to it as you want.

# TLDR
A `--output` command to export specific files during an image build run.
```sh
--output=</absolute/path/to/folder/in/image>:<layer> <relative/path/host/machine>
```

```sh
# several exports example
--output=</absolute/path/to/folder/in/image>:<layer> <relative/path/host/machine> 
--output=</absolute/path/to/folder/in/image>:<layer2> <relative/path/host/machine>
--output=</absolute/path/to/folder/in/image>:<layer3> <relative/path/host/machine>
--output=</absolute/path/to/folder/in/image> <relative/path/host/machine>  #last layer
```

**Comment by rufreakde** [comment]
Bumb any news?

**Comment by aaron-prindle** [comment]
@rufreakde +1 I think something similar to docker's `--output` flag would definitely help kaniko users.  Would you want to drive the creation of this feature?

**Comment by rufreakde** [comment]
@aaron-prindle I could try to look into this. I am not an expert but maybe I can create at least a draft that can be improved later on. Need to take some time out from my schedule for it.

Is there a good open source contribution guide for this repo I can read prior? Will check it when I have time.

**Comment by fireinmysou1** [comment]
> @aaron-prindle I could try to look into this. I am not an expert but maybe I can create at least a draft that can be improved later on. Need to take some time out from my schedule for it.
> 
> Is there a good open source contribution guide for this repo I can read prior? Will check it when I have time.

Hi there, any news?

**Comment by aaron-prindle** [comment]
@rufreakde Kaniko's contribution guides can be found here:
https://github.com/GoogleContainerTools/kaniko/blob/main/CONTRIBUTING.md
https://github.com/GoogleContainerTools/kaniko/blob/main/DEVELOPMENT.md

Let me know if you have any questions and if you are still are interested in working on this, thanks.
